src/cascadia/TerminalCore/TerminalSelection.cpp

```diff
@@ -263,3 +264,215 @@ void Terminal::ColorSelection(const COORD, const COORD, const TextAttribute)
 {
     THROW_HR(E_NOTIMPL);
 }
+
+// Method Description:
+// - update _selection's endpoint according to the direction provided.
+//    It is moved according the the expansion mode.
+// Arguments:
+// - dir: the direction that the selection endpoint will attempt to move to
+// - mode: the selection expansion mode that the selection anchor will adhere to
+// Return Value:
+// - <none>
+void Terminal::MoveSelectionAnchor(Direction dir, SelectionExpansionMode mode, SelectionEndpoint target)
+{
+    COORD endpoint = ((target == SelectionEndpoint::Start) ? _selection->start : _selection->end);

```

Comment:
If we made this a `COORD&`, then would we avoid needing to do the re-assignment at the bottom of this method?